clients/apply/source.go

```diff
@@ -0,0 +1,215 @@
+package apply
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/influxdata/influx-cli/v2/api"
+	"github.com/influxdata/influx-cli/v2/pkg/github"
+	"github.com/influxdata/influx-cli/v2/pkg/jsonnet"
+	"gopkg.in/yaml.v3"
+)
+
+type TemplateEncoding int
+
+const (
+	TemplateEncodingUnknown TemplateEncoding = iota
+	TemplateEncodingJson
+	TemplateEncodingJsonnet
+	TemplateEncodingYaml
+)
+
+func (e *TemplateEncoding) Set(v string) error {
+	switch v {
+	case "jsonnet":
+		*e = TemplateEncodingJsonnet
+	case "json":
+		*e = TemplateEncodingJson
+	case "yml", "yaml":
+		*e = TemplateEncodingYaml
+	default:
+		return fmt.Errorf("unknown inEncoding %q", v)
+	}
+	return nil
+}
+
+func (e TemplateEncoding) String() string {
+	switch e {
+	case TemplateEncodingJsonnet:
+		return "jsonnet"
+	case TemplateEncodingJson:
+		return "json"
+	case TemplateEncodingYaml:
+		return "yaml"
+	case TemplateEncodingUnknown:
+		fallthrough
+	default:
+		return "unknown"
+	}
+}
+
+type TemplateSource struct {
+	Name     string
+	Encoding TemplateEncoding
+	Open     func(context.Context) (io.ReadCloser, error)
+}
+
+func SourcesFromPath(path string, recur bool, encoding TemplateEncoding) ([]TemplateSource, error) {
+	paths, err := findPaths(path, recur)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find inputs at path %q: %w", path, err)
+	}
+
+	sources := make([]TemplateSource, len(paths))
+	for i := range paths {
+		path := paths[i] // Local var for the `Open` closure to capture.
+		encoding := encoding
+		if encoding == TemplateEncodingUnknown {
+			ext := filepath.Ext(path)
+			switch {
+			case strings.HasPrefix(ext, ".jsonnet"):
```

Comment:
Any reason this switch can't be something more like:

```
switch ext {
  case ".jsonnet"
    etc...
}
```

...just not sure why it has to be a `HasPrefix` check. Maybe there's some reason for that though!